### ISSUE TITLE
Gemma 4 E2B on Pixel 8: 0.10.0 fails on GPU decode, self-built 0.10.1 runtime patch fixes it

### ISSUE THREAD

**Issue by Shoolife** [google-ai-edge/LiteRT-LM#1850]
## Summary

I was testing `litert-community/gemma-4-E2B-it-litert-lm` on a `Pixel 8`.

With `litertlm-android 0.10.0`, the model already loads on `GPU`, but decode fails with:

- `Failed to clEnqueueNDRangeKernel - Invalid command queue`
- `Node number 2068 (LITERT_CL) failed to invoke`

I then built `LiteRT-LM 0.10.1` from source and patched the Android runtime path. With that patched runtime, the same model runs successfully on `Pixel 8 GPU` instead of failing on decode.

## Environment

- Device: `Pixel 8`
- OS: Android 16
- Model: `litert-community/gemma-4-E2B-it-litert-lm`
- App integration: Android app using `litertlm-android`

### `litertlm-android 0.9.0-alpha06`

`Gemma 4 E2B` did not get a real GPU path and immediately fell back / failed earlier in init.

### `litertlm-android 0.10.0`

This was already better:

- `LiteRT-LM model loaded: Gemma 4 E2B on GPU`
- `RunPrefillAsync` succeeds
- `RunDecodeAsync` starts

But decode crashes on OpenCL:

```text
Failed to clEnqueueNDRangeKernel - Invalid command queue
Node number 2068 (LITERT_CL) failed to invoke
```

I also checked app-level alternatives:

- `Session.generateContent(...)`
- synchronous `Conversation.sendMessage(...)`

Both hit the same decode crash, so this does not look like an app-level async/conversation issue.

## What fixed it in self-built `0.10.1`

I patched the runtime in a few places:

1. Android sampler loading
   - make Android load `libLiteRt.so` and `LiteRtTopKOpenClSampler` from app files

2. Shared library linkage
   - patched sampler `.so` to include `DT_NEEDED=libLiteRt.so`
   - removed the relevant export/versioning barrier so the sampler can resolve LiteRT runtime symbols

3. Gemma 4 GPU profile in JNI
   - the working config was:
     - `external_tensor_mode = false`
     - `hint_waiting_for_completion = true`

Important note:

- `external_tensor_mode = true` was **not** the fix
- it removed one crash, but then broke prefill on tensor-byte mismatch

The working combination was:

- `external_tensor_mode = false`
- `hint_waiting_for_completion = true`

## Key observation

Before the fix, the GPU logits path was landing on the old broken buffer profile.

After the fix, the logits requirements changed to a working packed GPU path, and the OpenCL sampler stopped failing.

That removed the decode failure and the model completed inference successfully on GPU.

## Successful GPU smoke result

After the runtime patch:

- `LiteRT-LM model loaded: Gemma 4 E2B on GPU`
- `RunPrefillAsync status: OK`
- `RunDecodeAsync` succeeds

Smoke metrics:

- `TTFT=2596 ms`
- `prefill=172 tok/s`
- `decode=11 tok/s`
- `tokens=21`
- `timeMs=4305`

## CPU vs GPU throughput after the runtime fix

Using the same model on the same device:

- CPU:
  - `prefill=53.35 tok/s`
  - `decode=10.57 tok/s`
- GPU with the patched self-built `LiteRT-LM 0.10.1` runtime:
  - `prefill=426.64 tok/s`
  - `decode=11.94 tok/s`

So after the fix the runtime is not only avoiding the crash, but is also clearly taking the intended GPU path for prefill.

## Request

Could you please advise:

1. whether this `Gemma 4 E2B` Android GPU decode problem is already known
2. whether the intended fix is closer to:
   - a default `Gemma 4` GPU runtime profile
   - OpenCL sampler / buffer-type handling
   - Android GPU delegate scheduling / waiting semantics
3. whether you would like a minimal patch diff attached here for reference

I can also provide:

- the exact patched files
- relevant log snippets
- a reduced repro flow
